### ISSUE TITLE
exec user process caused "exec format error"

### ISSUE THREAD

**Issue by namreg** [ceph/ceph-csi#253]
Hello there,

I am trying to deploy `rbdplugin` by running the command `examples/rbd/plugin-deploy.sh`. After that, I see the following status of the deploy:
```
NAME                          READY   STATUS             RESTARTS   AGE
csi-rbdplugin-287zp           2/2     Running            0          18m
csi-rbdplugin-attacher-0      1/1     Running            0          18m
csi-rbdplugin-provisioner-0   2/3     CrashLoopBackOff   8          18m
csi-rbdplugin-sfczc           2/2     Running            0          18m
csi-rbdplugin-tsw7h           2/2     Running            0          18m
```
As we can see, the provisioner status is `CrashLoopBackOff`. I found out container `rbdplugin` in that pod has failed.
```bash
kubectl logs csi-rbdplugin-provisioner-0  -c csi-rbdplugin
```
the output of the previous command is:
```bash
standard_init_linux.go:207: exec user process caused "exec format error"
```
AFAIK, it is docker error. Googling did not give me a result.

```
# docker version
Client:
 Version:           18.09.2
 API version:       1.39
 Go version:        go1.10.6
 Git commit:        3d8ffcd
 Built:             Thu Feb 14 11:26:37 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.2
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.6
  Git commit:       3d8ffcd
  Built:            Thu Feb 14 11:25:48 2019
  OS/Arch:          linux/amd64
  Experimental:     false

```

```
uname -a
Linux P20B-VR3-R1-K8S-N1 4.14.102.1 #1 SMP Thu Feb 21 16:01:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

```
kubernets version 1.12
```

Thanks.

**Comment by rootfs** [comment]
that's odd, csi-rbdplugin container looks running well in other pods. 

Can you get an output of kubectl get pods -o wide and see which host the provisioner was running and if that host has a different cpu?

**Comment by namreg** [comment]
@rootfs 
```
# kubectl get pods -o wide
NAME                          READY   STATUS             RESTARTS   AGE   IP              NODE                   NOMINATED NODE
csi-rbdplugin-26wzs           2/2     Running            0          27m   172.30.1.1      p20b-vr3-r1-ceph-ds1   <none>
csi-rbdplugin-4rlpj           2/2     Running            0          27m   172.30.100.2    p20b-vr3-r1-k8s-n2     <none>
csi-rbdplugin-attacher-0      1/1     Running            0          27m   10.43.244.212   p20b-vr3-r1-k8s-n2     <none>
csi-rbdplugin-provisioner-0   1/3     CrashLoopBackOff   19         27m   10.43.174.132   p20b-vr3-r1-k8s-n1     <none>
csi-rbdplugin-z7qzn           2/2     Running            0          27m   172.30.100.1    p20b-vr3-r1-k8s-n1     <none>

```

**Comment by rootfs** [comment]
which csi image you are using? 
cc @Madhu-1

**Comment by Madhu-1** [comment]
@namreg if you give some more information as @rootfs suggested

and I would also like to know the host image used in your deployment, I would try to reproduce the issue once I get this information, looks like some dependency issue for me.

**Comment by namreg** [comment]
I figured out what's the problem. We have `init` option on in the our docker daemon config file (the last param).
```bash
$ cat /etc/docker/daemon.json 
{
    "hosts": ["fd://", "tcp://127.0.0.1:4243", "unix:///run/docker.sock"],
    "registry-mirrors": ["http://10.3.0.15:5000"],
    "log-opts": {"max-size": "50m", "max-file": "2"},
    "log-driver":  "json-file",
    "storage-driver": "overlay2",
    "dns-opt": ["single-request"],
    "live-restore": true,
	"init": true
}
```
According to the docs this options responsible for the `Run an init in the container to forward signals and reap processes`. In our version of the docker the location of the tini (init system) is `/dev/init`. But for deploying the `provisioner` we re-mount `dev` directory (see https://github.com/ceph/ceph-csi/blob/csi-v1.0/deploy/rbd/kubernetes/csi-rbdplugin-provisioner.yaml#L87-L88).
The location of the tini was changed in the commit https://github.com/docker/docker-ce/commit/6b34fcbbd71facd54de64484a7ddfeec0bf19d6e#diff-f1e0be6b4f3bd67f34c801e147ece3b4

For my case I just mount docker init from host to /dev/init explicitly.
